backbone/src/main/java/org/researchstack/backbone/task/Task.java

```diff
@@ -40,6 +46,53 @@ public Task(String identifier)
     }
 
     /**
+     * Sets the current @{@link ViewTaskActivity} for this task.
+     * This function should be called only by the {@link ViewTaskActivity} instance running
+     * at its creation time.
+     * @param activity the @{@link ViewTaskActivity} instance
+     */
+    public void setActivity(ViewTaskActivity activity){
```

Comment:
PS: I my humble opinion, the whole task / ViewTaskActivity would need a radical rethink...